src/main/java/org/pegdown/Parser.java

```diff
@@ -774,39 +757,42 @@ public Rule Strong() {
         return NodeSequence( FirstOf( EmphOrStrong("**"), EmphOrStrong("__") ) );
     }
 
+    // vsch: TODO: test for unclosed strikethrough sequence carrying through the isClosed attribute, as soon as I can figure out how
     public Rule Strike() {
+        // vsch: we need to preserve isClosed() attribute of the StrongEmphSuperNode, otherwise we can have
+        // a strike sequence which is not closed and it makes it difficult to split out the lead-in/termination
+        // characters for syntax highlighting see: https://github.com/vsch/idea-multimarkdown for an example
         return NodeSequence(
                 EmphOrStrong("~~"),
-                push(new StrikeNode(popAsNode().getChildren()))
+                push(new StrikeNode((StrongEmphSuperNode) popAsNode()))
         );
     }
 
     @Cached
-    public Rule EmphOrStrong(String chars) {    
+    public Rule EmphOrStrong(String chars) {
         return Sequence(
                 Test(mayEnterEmphOrStrong(chars)),
                 EmphOrStrongOpen(chars),
-                push(new StrongEmphSuperNode(chars)),                
+                push(new StrongEmphSuperNode(chars)),
                 OneOrMore(
-                 TestNot(EmphOrStrongClose(chars)),
-                 Inline(),
-                 FirstOf(
-                  Sequence(
-                   //if current inline ends with a closing char for a current strong node: 
-                   Test(isStrongCloseCharStolen( chars )),
-                   //and composes a valid strong close:
-                   chars.substring(0, 1),
-                   //which is not followed by another closing char (e.g. in __strong _nestedemph___):
-                   TestNot(chars.substring(0, 1)),
-                   //degrade current inline emph to unclosed and mark current strong node for closing
-                   stealBackStrongCloseChar() 
-                  ),
-                  addAsChild()
-                 )
+                        TestNot(EmphOrStrongClose(chars)),
```

Comment:
Also, it looks like these are tabs, which I'd like to not appear in the source at all.